### DISCUSSION TITLE
How to return generic class from endpoint?

### DISCUSSION THREAD

**Discussion by detroitpro** [TypedProject/ts-express-decorators#1573] [Q&A] [answered]
In the following example, I am getting a compiler error when passing `ApiResult<UserResponse>` to the Returns decorator. Is there any way to return a generic class?

```javascript
  @Get('/:id')
  @Description('Get user')
  @(Returns(200, ApiResult<UserResponse>).ContentType('application/json')) //here is where I am getting an error
  async getUser(@PathParams('id') id: string): Promise<ApiResult<UserResponse>> {
    const user = await this.userService.getUser(id);
    if (user === undefined) {
      return ApiResult.error<UserResponse>(`unable to find user`);
    }
    return ApiResult.success<UserResponse>(user);
  }

```

**Comment by Romakita** [comment, marked as answer]
@detroitpro I invite you to read the documentation about generics ;)

https://tsed.io/docs/model.html#generics

Why this notation is incorrect?
```typescript
Returns(200, ApiResult<UserResponse>)
```
Written as it, ApiResult<UserResponse> is considered by typescript as an interface and not to a valid parameter (so a valid javascript object after transpilation)! 

The correct syntax could be:

```typescript
Returns(200, ApiResult).Of(UserResponse)
```
But you have to add extra decorator on ApiResult to propagate the UserResponse schema on this endpoint. Look the docs to found the answers ;)

See you
Romain
